src/main/java/wolox/training/authentication/SpringContextAuthentication.java

```diff
@@ -0,0 +1,17 @@
+package wolox.training.authentication;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SpringContextAuthentication implements IAuthentication {
+
+    @Override
+    public Authentication getAuthentication() {
+
+        return SecurityContextHolder.getContext().getAuthentication();
```

Comment:
You can use the method shown in the section `3. Get the User in a Controller` of the documentation which is less code and there is no need to set up this class, but anyway this is ok.